classes/Hook.php

```diff
@@ -63,6 +63,11 @@ class HookCore extends ObjectModel
 
     public static $native_module;
 
+    /**
+     * @var array|null Names of active hooks
+     */
+    protected static $active_hooks = null;
```

Comment:
You should add a line to clear this static field in the `resetStaticCache` method This will offer the possibility to clean it after modifications for example It's also very useful for our test environment where we frequently need to clear the cache